### ISSUE TITLE
[Bug]: Crash on edit profile

### ISSUE THREAD

**Issue by SimplyCorbett** [GeopJr/Tuba#1168]
### Describe the bug

When trying to edit your profile tuba instantly freezes/crashes

### Steps To Reproduce

Open profile -> edit profile ->freeze/crash

### Logs and/or Screenshots

```
simplycorbett@aragon ~ $ G_MESSAGES_DEBUG=Tuba flatpak run dev.geopjr.Tuba
(dev.geopjr.Tuba:2): Tuba-DEBUG: 04:05:57.747: Application.vala:247: os: GNOME 46 (Flatpak runtime)
(dev.geopjr.Tuba:2): Tuba-DEBUG: 04:05:57.747: Application.vala:247: prefix: /app
(dev.geopjr.Tuba:2): Tuba-DEBUG: 04:05:57.747: Application.vala:247: flatpak: true
(dev.geopjr.Tuba:2): Tuba-DEBUG: 04:05:57.747: Application.vala:247: version: 0.8.4 (production)
(dev.geopjr.Tuba:2): Tuba-DEBUG: 04:05:57.747: Application.vala:247: gtk: 4.14.5 (4.14.4)
(dev.geopjr.Tuba:2): Tuba-DEBUG: 04:05:57.747: Application.vala:247: libadwaita: 1.5.4 (1.5.3)
(dev.geopjr.Tuba:2): Tuba-DEBUG: 04:05:57.747: Application.vala:247: libsoup: 3.4.4 (3.4.4)
(dev.geopjr.Tuba:2): Tuba-DEBUG: 04:05:57.747: Application.vala:247: libgtksourceview: 5.12.1 (5.12.1)

(dev.geopjr.Tuba:2): GLib-GIO-WARNING **: 04:05:57.748: Can't find module 'dconf' specified in GSETTINGS_BACKEND
(dev.geopjr.Tuba:2): Tuba-DEBUG: 04:05:57.749: SecretAccountStore.vala:9: Using libsecret v0.21.4
(dev.geopjr.Tuba:2): Tuba-DEBUG: 04:05:57.765: Network.vala:65: GET: https://login.ready.tips/api/v1/accounts/9z49ipo3c6gsjz2f
Tuba-INFO: 04:05:57.765: Streams.vala:88: TubaMastodonAccount > https://login.ready.tips/api/v1/streaming?stream=user:notification
(dev.geopjr.Tuba:2): Tuba-DEBUG: 04:05:57.765: Streams.vala:68: Opening stream: https://login.ready.tips/api/v1/streaming?stream=user:notification
(dev.geopjr.Tuba:2): Tuba-DEBUG: 04:05:57.765: Network.vala:65: GET: https://login.ready.tips/api/v1/markers?timeline[]=notifications
(dev.geopjr.Tuba:2): Tuba-DEBUG: 04:05:57.766: Network.vala:65: GET: https://mastodon.social/api/v1/accounts/112006313583452831
Tuba-INFO: 04:05:57.766: Streams.vala:88: TubaMastodonAccount > https://mastodon.social/api/v1/streaming?stream=user:notification
(dev.geopjr.Tuba:2): Tuba-DEBUG: 04:05:57.766: Streams.vala:68: Opening stream: https://mastodon.social/api/v1/streaming?stream=user:notification
(dev.geopjr.Tuba:2): Tuba-DEBUG: 04:05:57.766: Network.vala:65: GET: https://mastodon.social/api/v1/markers?timeline[]=notifications
(dev.geopjr.Tuba:2): Tuba-DEBUG: 04:05:57.766: Network.vala:65: GET: https://login.ready.pics/api/v1/accounts/9z46ud7isvthne2f
Tuba-INFO: 04:05:57.766: Streams.vala:88: TubaMastodonAccount > https://login.ready.pics/api/v1/streaming?stream=user:notification
(dev.geopjr.Tuba:2): Tuba-DEBUG: 04:05:57.766: Streams.vala:68: Opening stream: https://login.ready.pics/api/v1/streaming?stream=user:notification
(dev.geopjr.Tuba:2): Tuba-DEBUG: 04:05:57.766: Network.vala:65: GET: https://login.ready.pics/api/v1/markers?timeline[]=notifications
(dev.geopjr.Tuba:2): Tuba-DEBUG: 04:05:57.766: Network.vala:65: GET: https://login.ready.social/api/v1/accounts/9z2inen467w827b8
Tuba-INFO: 04:05:57.766: Streams.vala:88: TubaMastodonAccount > https://login.ready.social/api/v1/streaming?stream=user:notification
(dev.geopjr.Tuba:2): Tuba-DEBUG: 04:05:57.766: Streams.vala:68: Opening stream: https://login.ready.social/api/v1/streaming?stream=user:notification
(dev.geopjr.Tuba:2): Tuba-DEBUG: 04:05:57.766: Network.vala:65: GET: https://login.ready.social/api/v1/markers?timeline[]=notifications
(dev.geopjr.Tuba:2): Tuba-DEBUG: 04:05:57.766: Network.vala:65: GET: https://login.ready.lgbt/api/v1/accounts/9z4nz84mpzh62f0g
Tuba-INFO: 04:05:57.766: Streams.vala:88: TubaMastodonAccount > https://login.ready.lgbt/api/v1/streaming?stream=user:notification
(dev.geopjr.Tuba:2): Tuba-DEBUG: 04:05:57.766: Streams.vala:68: Opening stream: https://login.ready.lgbt/api/v1/streaming?stream=user:notification
(dev.geopjr.Tuba:2): Tuba-DEBUG: 04:05:57.766: Network.vala:65: GET: https://login.ready.lgbt/api/v1/markers?timeline[]=notifications
(dev.geopjr.Tuba:2): Tuba-DEBUG: 04:05:57.766: Network.vala:65: GET: https://login.ready.art/api/v1/accounts/9z461n2268v19b5p
Tuba-INFO: 04:05:57.766: Streams.vala:88: TubaMastodonAccount > https://login.ready.art/api/v1/streaming?stream=user:notification
(dev.geopjr.Tuba:2): Tuba-DEBUG: 04:05:57.766: Streams.vala:68: Opening stream: https://login.ready.art/api/v1/streaming?stream=user:notification
(dev.geopjr.Tuba:2): Tuba-DEBUG: 04:05:57.766: Network.vala:65: GET: https://login.ready.art/api/v1/markers?timeline[]=notifications
(dev.geopjr.Tuba:2): Tuba-DEBUG: 04:05:57.766: SecretAccountStore.vala:92: Loaded 6 accounts
(dev.geopjr.Tuba:2): Tuba-DEBUG: 04:05:57.766: AccountStore.vala:85: Activating @simplycorbett@login.ready.social…
(dev.geopjr.Tuba:2): Tuba-DEBUG: 04:05:57.766: Network.vala:65: GET: https://login.ready.social/api/v1/accounts/verify_credentials
(dev.geopjr.Tuba:2): Tuba-DEBUG: 04:05:57.766: Network.vala:65: GET: https://login.ready.social/api/v1/instance
(dev.geopjr.Tuba:2): Tuba-DEBUG: 04:05:57.766: Network.vala:65: GET: https://login.ready.social/api/v1/custom_emojis
(dev.geopjr.Tuba:2): Tuba-DEBUG: 04:05:57.766: Network.vala:65: GET: https://login.ready.social/api/v1/announcements
(dev.geopjr.Tuba:2): Tuba-DEBUG: 04:05:57.766: Application.vala:434: Presenting MainWindow
Tuba-INFO: 04:05:57.810: Streams.vala:88: TubaViewsHome > https://login.ready.social/api/v1/streaming?stream=user
(dev.geopjr.Tuba:2): Tuba-DEBUG: 04:05:57.810: Streams.vala:68: Opening stream: https://login.ready.social/api/v1/streaming?stream=user
Tuba-INFO: 04:05:57.811: Streams.vala:94: TubaViewsHome X https://login.ready.social/api/v1/streaming?stream=user
(dev.geopjr.Tuba:2): Tuba-DEBUG: 04:05:57.811: Streams.vala:99: Closing: https://login.ready.social/api/v1/streaming?stream=user
Tuba-INFO: 04:05:57.811: Streams.vala:88: TubaViewsHome > https://login.ready.social/api/v1/streaming?stream=user
(dev.geopjr.Tuba:2): Tuba-DEBUG: 04:05:57.811: Streams.vala:68: Opening stream: https://login.ready.social/api/v1/streaming?stream=user
Tuba-INFO: 04:05:57.812: Streams.vala:88: TubaViewsNotifications > https://login.ready.social/api/v1/streaming?stream=user:notification
Tuba-INFO: 04:05:57.812: Streams.vala:94: TubaViewsNotifications X https://login.ready.social/api/v1/streaming?stream=user:notification
Tuba-INFO: 04:05:57.812: Streams.vala:88: TubaViewsNotifications > https://login.ready.social/api/v1/streaming?stream=user:notification
Tuba-INFO: 04:05:57.813: Streams.vala:88: TubaViewsConversations > https://login.ready.social/api/v1/streaming?stream=direct
(dev.geopjr.Tuba:2): Tuba-DEBUG: 04:05:57.813: Streams.vala:68: Opening stream: https://login.ready.social/api/v1/streaming?stream=direct
Tuba-INFO: 04:05:57.813: Streams.vala:94: TubaViewsConversations X https://login.ready.social/api/v1/streaming?stream=direct
(dev.geopjr.Tuba:2): Tuba-DEBUG: 04:05:57.813: Streams.vala:99: Closing: https://login.ready.social/api/v1/streaming?stream=direct
Tuba-INFO: 04:05:57.813: Streams.vala:88: TubaViewsConversations > https://login.ready.social/api/v1/streaming?stream=direct
(dev.geopjr.Tuba:2): Tuba-DEBUG: 04:05:57.813: Streams.vala:68: Opening stream: https://login.ready.social/api/v1/streaming?stream=direct
(dev.geopjr.Tuba:2): Tuba-DEBUG: 04:05:57.991: Network.vala:65: GET: https://login.ready.social/api/v1/timelines/home?limit=20
(dev.geopjr.Tuba:2): Tuba-DEBUG: 04:05:57.991: Network.vala:65: GET: https://login.ready.social/api/v1/notifications?limit=20
(dev.geopjr.Tuba:2): Tuba-DEBUG: 04:05:57.991: Network.vala:65: GET: https://login.ready.social/api/v1/conversations?limit=20

(dev.geopjr.Tuba:2): Tuba-WARNING **: 04:05:58.299: Image.vala:69: Failed to download image at "https://login.ready.tips/files/public-7c36aa25-726d-4aaa-99cc-1231016e9509.webp": Server returned Not Found

(dev.geopjr.Tuba:2): Tuba-WARNING **: 04:05:58.317: Streams.vala:78: Error opening stream: The server did not accept the WebSocket handshake.

(dev.geopjr.Tuba:2): Tuba-WARNING **: 04:05:58.390: Streams.vala:78: Error opening stream: The server did not accept the WebSocket handshake.

(dev.geopjr.Tuba:2): Tuba-WARNING **: 04:05:58.394: Streams.vala:78: Error opening stream: The server did not accept the WebSocket handshake.

(dev.geopjr.Tuba:2): Tuba-WARNING **: 04:05:58.396: Streams.vala:78: Error opening stream: The server did not accept the WebSocket handshake.

(dev.geopjr.Tuba:2): Tuba-WARNING **: 04:05:58.396: Streams.vala:78: Error opening stream: The server did not accept the WebSocket handshake.

(dev.geopjr.Tuba:2): Tuba-WARNING **: 04:05:58.397: Streams.vala:78: Error opening stream: The server did not accept the WebSocket handshake.

(dev.geopjr.Tuba:2): Tuba-WARNING **: 04:05:58.445: Streams.vala:78: Error opening stream: The server did not accept the WebSocket handshake.

(dev.geopjr.Tuba:2): Tuba-WARNING **: 04:05:58.445: Streams.vala:78: Error opening stream: The server did not accept the WebSocket handshake.

(dev.geopjr.Tuba:2): Tuba-WARNING **: 04:05:58.446: Streams.vala:78: Error opening stream: The server did not accept the WebSocket handshake.

(dev.geopjr.Tuba:2): Tuba-CRITICAL **: 04:05:58.464: Network.vala:83: Request "https://login.ready.tips/api/v1/accounts/9z49ipo3c6gsjz2f" failed: 404 Not Found

(dev.geopjr.Tuba:2): Tuba-CRITICAL **: 04:05:58.464: Network.vala:83: Request "https://login.ready.tips/api/v1/markers?timeline[]=notifications" failed: 404 Not Found
(dev.geopjr.Tuba:2): Tuba-DEBUG: 04:05:58.596: Network.vala:65: GET: https://mastodon.social/api/v1/notifications?min_id=326619720
(dev.geopjr.Tuba:2): Tuba-DEBUG: 04:05:58.604: InstanceAccount.vala:200: @simplycorbett@login.ready.social: profile updated
(dev.geopjr.Tuba:2): Tuba-DEBUG: 04:06:01.512: Network.vala:65: GET: https://login.ready.social/api/v1/accounts/relationships?id=9z2inen467w827b8
(dev.geopjr.Tuba:2): Tuba-DEBUG: 04:06:01.517: Network.vala:65: GET: https://login.ready.social/api/v1/accounts/9z2inen467w827b8/statuses?only_media=false&exclude_replies=true&limit=20
(dev.geopjr.Tuba:2): Tuba-DEBUG: 04:06:01.517: Network.vala:65: GET: https://login.ready.social/api/v1/accounts/9z2inen467w827b8/statuses?pinned=true
```

### Instance Backend

I'm not sure / Other

### Operating System

Gentoo

### Package

Flatpak

### Troubleshooting information

os: GNOME 46 (Flatpak runtime)
prefix: /app
flatpak: true
version: 0.8.4 (production)
gtk: 4.14.5 (4.14.4)
libadwaita: 1.5.4 (1.5.3)
libsoup: 3.4.4 (3.4.4)
libgtksourceview: 5.12.1 (5.12.1)

### Additional Context

I'm not sure why its trying to load ready.tips as I removed that instance from tuba (and that instance is down) so...

Running on iceshrimp.net

**Comment by GeopJr** [comment]
For starters, let's fix the `Can't find module 'dconf' specified in GSETTINGS_BACKEND` error. Do app settings even get saved on your machine? Might explain the deleted account issue.

It's a host issue, closest I can find is this https://bugs.gentoo.org/878043

For the deleted account part, open `seahorse` (password and keys), kwallet or any libsecret manager and remove the deleted account from your keyring. If it's not there, try key rack as it's flatpak-first https://flathub.org/apps/app.drey.KeyRack

I'll take another look on why the deleted account might have been added to the accounts internally but not on the UI? (It wasn't on the UI/account switcher, was it?)

As for the profile editor crash, I recently found and fixed a GTK issue I came across with it, not sure if it's the same. I also need to check iceshrimp.net to make sure it implements everything as expected, but it shouldn't crash if it didn't anyway :/

Would you mind getting a backtrace? It's quite tedious so feel free to do it whenever you are free:


- `$ flatpak install runtime/org.gnome.Sdk` (if tuba was installed for the current user only, pass --user too)
- From the list, select `runtime/org.gnome.Sdk/<arch>/46` (<-- **46**)
- `$ flatpak install dev.geopjr.Tuba.Debug` (if tuba was installed for the current user only, pass --user too)
- `$ flatpak run --command=sh --devel dev.geopjr.Tuba`
- `[📦 dev.geopjr.Tuba ~]$ gdb /app/bin/dev.geopjr.Tuba`
- `(gdb) run`
- Reproduce it (make sure it crashes!)
- `(gdb) backtrace`
- Send back the log
- Feel free to remove the packages you installed above

**Comment by GeopJr** [comment]
> I mean the accounts save and are there on restart, so yes.

No, those are saved in your keyring, app settings are saved in dconf. E.g. if you disable blurhashes, close the app and re-open it, is "Blurhash" disabled?

> The deleted account was on the UI until I deleted it. Unfortunately, it's not in seahorse. I will try keyrack when I have the time.

The more I look at it the more confused I am. Could you try again just to make sure? Run tuba from terminal, make sure you see it attempt to connect to ready.tips and make sure there's no ready.tips accounts in the account switcher.

Otherwise, you may try to remove it using secret-tool, but then again if seahorse did not show the ready.tips account, I doubt secre-tool will be able to see it either. Maybe it's stuck in some weird limbo or the keyring you are using doesn't play nicely with flatpak :shrug: Anyway `secret-tool clear xdg:schema dev.geopjr.Tuba login @USERNAME@login.ready.tips`

Hopefully key rack can see it :/

> `simplycorbett@aragon ~ $ flatpak install runtime/org.gnome.sdk --user`

It's case sensitive, **S**dk

> I've only recently started to use flatpak and I'm fairly unfamiliar with it. Will check into this when I get the free time.

Don't worry about it, thanks for helping me with debugging :)

**Comment by GeopJr** [comment]
That's really helpful, thanks!

Quick question, was your Gentoo machine also aarch64? Because if it wasn't and you didn't see the same logs (`Too many GWeakRef registered`), it might have been a different bug :/ Either way, the bug in the logs above should be fixed in #1175, thanks!
